liblouis/compileTranslationTable.c

```diff
@@ -788,6 +792,20 @@ passFindCharacters(const FileInfo *file, widechar *instructions, int end,
 	goto NO_CHARACTERS;
 }
 
+static const char *
+printSource(const FileInfo *currentFile, const char *sourceFile, int sourceLine) {
+	static char scratchBuf[MAXSTRING];
+	if (sourceFile) {
+		if (currentFile && currentFile->sourceFile &&
+				strcmp(currentFile->sourceFile, sourceFile) == 0)
+			snprintf(scratchBuf, MAXSTRING, "line %d", sourceLine);
+		else
+			snprintf(scratchBuf, MAXSTRING, "%s:%d", sourceFile, sourceLine);
```

Comment:
Because `printSource()` is sometimes used to refer to other rules inside warning and error messages, which already start with the file name and line number of the "current" rule. I just thought that it would be nice to not print the file name twice if it's the same.